### PR TITLE
Purge aad

### PR DESCRIPTION
**Pull request by petrkotas**
Enhance the purge script with the ability to delete apps and role assignments.

/cc @mjudeikis

### CODE REVIEW

File: pkg/util/purge/purge.go [hunk 1]
```diff
@@ -27,19 +35,36 @@ type ResourceCleaner struct {
...
+	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	tenantID := os.Getenv("AZURE_TENANT_ID")
+
 	authorizer, err := auth.NewAuthorizerFromEnvironment()
 	if err != nil {
 		log.Fatal(err)
 	}
 
-	return &ResourceCleaner{
+	config := auth.NewClientCredentialsConfig(clientID, clientSecret, tenantID)
```
Comment (mjudeikis) [inline pkg/util/purge/purge.go:58]: `NewAuthorizerFromEnvironmentWithResource`

File: pkg/util/purge/purge.go [hunk 2]
```diff
@@ -27,19 +35,36 @@ type ResourceCleaner struct {
 	privatelinkservicescli network.PrivateLinkServicesClient
 	securitygroupscli      network.SecurityGroupsClient
```
Comment (mjudeikis) [inline pkg/util/purge/purge.go:37]: nit: no space.

File: hack/clean/clean.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+		if err != nil {
+			return err
+		}
+	} else {
+		ttl = defaultTTL
+	}
+
+	var createdTag = defaultCreatedAtTag
+	if os.Getenv("AZURE_PURGE_CREATED_TAG") != "" {
+		createdTag = os.Getenv("AZURE_PURGE_CREATED_TAG")
```
Comment (olga-mir) [inline hack/clean/clean.go:55]: you can avoid getting env twice by using LookupEnv:
`createdTag, found := os.LookupEnv("AZURE_PURGE_CREATED_TAG")`

**Comment by mjudeikis** [comment]
@petrkotas we should be good now to get this running again

File: pkg/util/purge/activedirectory.go [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+				filteredRoleAssignments = append(filteredRoleAssignments, rs)
+				rc.log.Debugf("RG: %s, roleAssignemnt: %s\n", rg, rs)
+			}
+		}
+	}
+
+	return filteredRoleAssignments, nil
+}
+
+func (rc *ResourceCleaner) cleanAAD(ctx context.Context, roleAssignments []authorization.RoleAssignment) error {
```
Comment (mjudeikis) [inline pkg/util/purge/activedirectory.go:64]: Can we have a behaviour documented as part of function documentations?
Like checking back relationship, AD ID URL, cluster existence. Basic logical flow explained.

Comment (petrkotas) [reply pkg/util/purge/activedirectory.go:124]: will this do?

**Comment by jim-minter** [comment]
This PR either needs to be got into a working state, or it should be closed.

**Comment by mjudeikis** [comment]
Closing this for now. lets discuss in 121

**Pull request closed without merging**